internal/signalfx-agent/pkg/core/agent.go

```diff
@@ -1,262 +0,0 @@
-// Package core contains the central frame of the agent that hooks up the
-// various subsystems.
-package core
-
-import (
-	"context"
-	"io"
-	"net/http"
-	"os"
-	"time"
-
-	log "github.com/sirupsen/logrus"
-
-	"github.com/signalfx/golib/v3/datapoint"
-	"github.com/signalfx/golib/v3/event"
-	"github.com/signalfx/golib/v3/trace"
-
-	"github.com/signalfx/signalfx-agent/pkg/core/config"
-	"github.com/signalfx/signalfx-agent/pkg/core/hostid"
-	"github.com/signalfx/signalfx-agent/pkg/core/meta"
-	"github.com/signalfx/signalfx-agent/pkg/core/services"
-	"github.com/signalfx/signalfx-agent/pkg/core/writer"
-	"github.com/signalfx/signalfx-agent/pkg/core/writer/tracetracker"
-	"github.com/signalfx/signalfx-agent/pkg/monitors"
-	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
-	"github.com/signalfx/signalfx-agent/pkg/observers"
-	"github.com/signalfx/signalfx-agent/pkg/utils"
-)
-
-//go:generate sh -c "test -e `go env GOPATH`/bin/genny > /dev/null || (cd ../.. && go install github.com/mauricelam/genny)"
```

Comment:
This is where genny is installed, which is expected by the splunk writer and causing the failed lint job. It can be relocated there, or I suspect that writer code can be removed since I don't believe it's exercised.

Reply:
@rmfitzpatrick thank you so much. I was tearing my head apart. Let me remove writer code and run tests to make sure everything is sane.